dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdSymbolFileUploadTask.kt

```diff
@@ -0,0 +1,159 @@
+package com.datadog.gradle.plugin
+
+import com.android.build.gradle.api.ApplicationVariant
+import com.android.build.gradle.tasks.ExternalNativeBuildTask
+import com.datadog.gradle.plugin.internal.ApiKey
+import com.datadog.gradle.plugin.internal.Uploader
+import org.gradle.api.Project
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.logging.Logging
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.ProviderFactory
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.TaskProvider
+import java.io.File
+import javax.inject.Inject
+import kotlin.reflect.full.memberProperties
+
+/**
+ * A Gradle task to upload NDK symbol files to Datadog servers.
+ */
+internal abstract class DdSymbolFileUploadTask @Inject constructor(
+    objects: ObjectFactory,
+    providerFactory: ProviderFactory
+) : DdFileUploadTask(providerFactory) {
+
+    @get:InputFiles
+    val searchDirectories: ConfigurableFileCollection = objects.fileCollection()
```

Comment:
I see that this property is initialized at registration. I wonder why do you also have to initialize it by default from constructor property. Is this a gradle task requirement ?

Reply:
I think so yes, but I'm a bit out of my element with Gradle on this front.

From what I can tell I can't inject the completed files collection, and I can only create one from the ObjectFactory, and I don't want it to be null.  So I create it during initialization from the injected ObjectFactory, then fill it as part of Register.

Reply:
Yes you cannot inject but maybe you can put it as `lateinit` and not null instead without having to initialize it.

Reply:
also I think what you nee there is more this lazy initialization of properties:
https://docs.gradle.org/current/userguide/lazy_configuration.html